action/protocol/staking/vote_reviser.go

```diff
@@ -96,10 +96,24 @@ func (vr *VoteReviser) calculateVoteWeight(csm CandidateStateManager) (Candidate
 		}
 
 		if cand.SelfStakeBucketIdx == bucket.Index {
-			cand.AddVote(calculateVoteWeight(vr.c, bucket, true))
+			err = cand.AddVote(calculateVoteWeight(vr.c, bucket, true))
```

Comment:
```
func (d *Candidate) AddVote(amount *big.Int) error {
	if amount.Sign() < 0 {
		return ErrInvalidAmount
	}
	d.Votes.Add(d.Votes, amount)
	return nil
}
```
if amount < 0, it does nothing, so I think we can simply ignore the error
```
_ = cand.AddVote(calculateVoteWeight(vr.c, bucket, true))
```

Reply:
I think if `return ErrInvalidAmount`, don't execute  Line107
`cand.SelfStake = bucket.StakedAmount`
so added featureCtx check

Reply:
right, discussed offline